File: day-02/PartTwo.cs
// SPDX-FileCopyrightText: 2020 Pablo Jiménez Pascual <pablo@jimpas.me>
//
// SPDX-License-Identifier: GPL-3.0-or-later

using System;
using System.Linq;

namespace day_02
{
    class PartTwo
    {
        static void Main(string[] args)
        {
            char[] delimiters = { '-', ' ', ':' };
            int cont = 0;
            string[] lines = System.IO.File.ReadAllLines(args[0]);
            foreach (string line in lines)
            {
                string[] data = line.Split(delimiters);
                int min = Int32.Parse(data[0]) - 1;
                int max = Int32.Parse(data[1]) - 1;
                char target = char.Parse(data[2]);
                string text = data[4];
                int freq = 0;
                if (text[min] == target)
                {
                    freq++;
                }
                if (text[max] == target)
                {
                    freq++;
                }
                if (freq == 1)
                {
                    cont++;
                }
            }
            Console.WriteLine(cont);
        }
    }
}

File: 2020/day-06/part_two.py
# SPDX-FileCopyrightText: 2020 Pablo Jiménez Pascual <pablo@jimpas.me>
#
# SPDX-License-Identifier: GPL-3.0-or-later

import sys

with open(sys.argv[1], "r") as file:
    letters_seen = dict()
    counts = list()
    n_members_group = 0
    for line in file:
        if not line.strip():
            cont = 0
            for l in letters_seen:
                if letters_seen[l] == n_members_group:
                    cont = cont + 1
            counts.append(cont)
            letters_seen = dict()
            n_members_group = 0
            continue
        n_members_group = n_members_group + 1
        for c in line.strip():
            if c in letters_seen:
                letters_seen[c] = letters_seen[c] + 1
            else:
                letters_seen[c] = 1
    cont = 0
    for l in letters_seen:
        if letters_seen[l] == n_members_group:
            cont = cont + 1
    counts.append(cont)
    print(sum(counts))

